integration_tests/gravity_utils.py

```diff
@@ -0,0 +1,188 @@
+
+    # deploy gravity contract to geth
+    gravity_id = cli.query_gravity_params()["params"]["gravity_id"]
+    signer_set = cli.query_latest_signer_set_tx()["signer_set"]["signers"]
+    powers = [int(signer["power"]) for signer in signer_set]
+    threshold = int(2**32 * 0.66)  # gravity normalize the power to [0, 2**32]
+    eth_addresses = [signer["ethereum_address"] for signer in signer_set]
+    assert sum(powers) >= threshold, "not enough validator on board"
+
+    contract = deploy_contract(
+        w3,
+        CONTRACTS["Gravity"],
+        (gravity_id.encode(), threshold, eth_addresses, powers, admin.address),
+    )
+    print("gravity contract deployed", contract.address)
+
+    # make all the orchestrator "Relayer" roles
+    k_relayer = sha3.keccak_256()
+    k_relayer.update(b"RELAYER")
+    for _, address in enumerate(eth_addresses):
+        set_role_tx = contract.functions.grantRole(
+            k_relayer.hexdigest(), address
+        ).build_transaction({"from": admin.address})
+        set_role_receipt = send_transaction(w3, set_role_tx, admin.key)
+        print("set_role_tx", set_role_receipt)
+
+    # start orchestrator:
+    # a) add process into the supervisord config file
+    # b) reload supervisord
+    programs = {}
+    for i, val in enumerate(custom_cronos.config["validators"]):
+        # update gravity contract in gorc config
+        gorc_config_path = custom_cronos.base_dir / f"node{i}/gorc.toml"
+        update_gravity_contract(gorc_config_path, contract.address)
+
+        programs[f"program:{chain_id}-orchestrator{i}"] = {
+            "command": (
+                f'gorc -c "{gorc_config_path}" orchestrator start '
+                "--cosmos-key cronos --ethereum-key eth --mode AlwaysRelay"
+            ),
+            "environment": "RUST_BACKTRACE=full",
+            "autostart": "true",
+            "autorestart": "true",
+            "startsecs": "3",
+            "redirect_stderr": "true",
+            "stdout_logfile": f"%(here)s/orchestrator{i}.log",
+        }
+
+    add_ini_sections(custom_cronos.base_dir / "tasks.ini", programs)
+    custom_cronos.supervisorctl("update")
+
+    yield GravityBridge(custom_cronos, w3, contract)
+
+
+def setup_cosmos_erc20_contract(cluster, denom, symbol):
+    # Create cosmos erc20 contract
+    print("Deploy cosmos erc20 contract on ethereum")
+    tx_receipt = deploy_erc20(
+        cluster.contract, cluster.geth, denom, denom, symbol, 6, KEYS["validator"]
+    )
+    assert tx_receipt.status == 1, "should success"
+    # Wait enough for orchestrator to relay the event
+    cronos_cli = cluster.cronos.cosmos_cli()
+    wait_for_new_blocks(cronos_cli, 30)
+    # Check mapping is done on cluster side
+    cosmos_erc20 = cronos_cli.query_gravity_contract_by_denom(denom)
+    print("cosmos_erc20:", cosmos_erc20)
+    assert cosmos_erc20 != ""
+    cosmos_erc20_contract = get_contract(
+        cluster.geth, cosmos_erc20["erc20"], CONTRACTS["TestERC21Source"]
```

Comment:
This is a mistake on the original code but this line may be confusing.

The ABI should be an `CosmosERC20` (https://github.com/crypto-org-chain/cronos/blob/main/integration_tests/contracts/contracts/CosmosToken.sol) not `TestERC21Source`

Need to add it in https://github.com/crypto-org-chain/cronos/blob/main/integration_tests/utils.py#L39